ansible/tasks/db/recreateDoc.yml

```diff
@@ -19,28 +19,36 @@
     existingDoc: "{{ response['content']|from_json }}"
   when: response.status == 200
 
+- set_fact:
+    create: False
+    newDocWithRevision: {}
+
 - name: extract revision from previous document
   vars:
     revision: "{{ existingDoc['_rev'] }}"
   set_fact:
     newDocWithRevision: "{{ doc | combine({'_rev': revision}) }}"
   when: response.status == 200
 
-- set_fact:
-    create: False
-
 - name: check if a doc update is required
   set_fact:
     create: True
   when: (response.status == 200 and existingDoc != newDocWithRevision) or response.status == 404 # Create doc, if it did not exist before or if update is required.
 
+- set_fact:
+    document: "{{ doc }}"
+
+- set_fact:
+    document: "{{ newDocWithRevision }}"
+  when: newDocWithRevision['_id'] is defined
+
 - name: recreate or update the document on the {{ dbName }} database
   uri:
     url: "{{ db_protocol }}://{{ db_host }}:{{ db_port }}/{{ dbName }}"
     method: POST
     status_code: 200, 201
     body_format: json
-    body: "{{ newDocWithRevision | default(doc) }}"
+    body: "{{ document }}"
```

Comment:
what happens here when `doc['_id']` is not defined? It is assumed implicitly by the script that `_id` is given in the document. we might want to check if that condition is true at the beginning of the task instead of failing somewhere?